utils/certs.go

```diff
@@ -5,45 +5,42 @@ import (
 	"crypto/tls"
 	"fmt"
 	"io/ioutil"
-	"net"
 	"net/http"
 	"path"
 	"strings"
 	"time"
 
+	"github.com/18F/cf-cdn-service-broker/config"
 	"github.com/aws/aws-sdk-go/aws"
 	"github.com/aws/aws-sdk-go/service/s3"
-	"github.com/xenolf/lego/acme"
-
-	"github.com/18F/cf-cdn-service-broker/config"
+	"github.com/go-acme/lego/certcrypto"
+	"github.com/go-acme/lego/lego"
+	"github.com/go-acme/lego/registration"
 )
 
-func preCheckDNS(fqdn, value string) (bool, error) {
-	record, err := net.LookupTXT(fqdn)
-	if err != nil {
-		return false, err
-	}
-	if len(record) == 1 && record[0] == value {
-		return true, nil
-	}
-	return false, fmt.Errorf("DNS precheck failed on name %s, value %s", fqdn, value)
-}
-
-func init() {
-	acme.PreCheckDNS = preCheckDNS
-}
+//func preCheckDNS(fqdn, value string) (bool, error) {
+//	record, err := net.LookupTXT(fqdn)
+//	if err != nil {
+//		return false, err
+//	}
+//	if len(record) == 1 && record[0] == value {
+//		return true, nil
+//	}
+//	return false, fmt.Errorf("DNS precheck failed on name %s, value %s", fqdn, value)
+//}
 
 type User struct {
-	Email        string
-	Registration *acme.RegistrationResource
+	Email string
+	// todo (mxplusb): fix this soon!
```

Comment:
this `todo` should be more descriptive

Reply:
in ides it will say _deprecated_ but I cleaned the comment up.